feature/bgp/gracefulrestart/ate_tests/bgp_graceful_restart_test/README.md

```diff
@@ -2,53 +2,141 @@
+
+
+## Topology
+Create the following connections:
+```mermaid
+graph LR; 
+A[ATE:Port1] -- EBGP --> B[Port1:DUT:Port2];
+B -- IBGP --> C[Port2:ATE];
+```
 
 ## Procedure
 
-*   Establish eBGP sessions between:
-    *   ATE port-1 and DUT port-1
-    *   ATE port-2 and DUT port-2
-    *   Configure allow route-policy under BGP peer-group address-family
+**RT-1.4.1: Enable and validate BGP Graceful restart feature**
+*   Configure EBGP peering between ATE:Port1 and DUT:Port1
+*   Configure IBGP peering between ATE:Port2 and DUT:Port2
+*   Ensure that the EBGP and IBGP peering are setup for IPv4-Unicast and IPv6-unicast AFI-SAFIs. Total 2xpeer-groups (1 per protocol) with 1 BGP session each.  
+*   Enable `Graceful-Restart` capability at the `Peer-Group` level.
+*   Ensure that the `restart-time` and the `stale-routes-time` are configured at the `Global` level
+*   Configure allow route-policy under BGP peer-group address-family
 *   Validate received capabilities at DUT and ATE reflect support for graceful
-    restart.
-*   For IPv4 and IPv6 routes:
-    *   (Restarting speaker) Advertise prefixes between the ATE ports, through
-        the DUT. Trigger DUT session restart by disconnecting TCP session
-        between DUT and ATE (this may be achieved by using an ACL), determine
-        that packets are:
-        *   Forwarded between ATE port-1 and DUT port-1 for the duration of the
-            specified stale routes time.
-        *   Dropped after the stale routes timer has expired.
-        *   Forwarded again between ATE port-1 and DUT port-1 after the session
-            is re-established.
-    *   (Receiving speaker) Advertise prefixes between the ATE ports, through
-        the DUT. Trigger session restart by disconnecting the BGP session from
-        ATE port-2.
-        *   Ensure that prefixes are propagated to ATE port-2 during the
-            restart.
-        *   Ensure that traffic can be forwarded between ATE port-1 and ATE
-            port-2 during stale routes time.
-        *   Ensure that prefixes are withdrawn, and traffic cannot be forwarded
-            between ATE port-1 and port-2 after the stale routes time expires.
+     restart.
+    
+
+**RT-1.4.2: Restarting DUT speaker whose BGP process was killed gracefully**
+*   Advertise prefixes between the ATE ports, through the DUT. 
+*   Trigger DUT session restart by killing the BGP process in the DUT. Please use the `gNOI.killProcessRequest_Signal_Term` as per [gNOI_proto](https://github.com/openconfig/gnoi/blob/main/system/system.proto#L326).
+     *   Please kill the right process to restart BGP. For Juniper it is the `RPD` process. For Arista and Cisco this is the `BGP` process. For Nokia this is `sr_bgp_mgr`.
+     *   Once the process is killied, verify that the packets are:
+          *   Forwarded between ATE port-1 and DUT port-1 for the duration of the specified stale routes time.
```

Comment:
add step to stop traffic.  Verify zero packet loss.

Reply:
Done